File: week02/이정현/SWEA_27024_줄서기앞뒤삽입.py
T = int(input())
# 노드 생서
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

for test_case in range(1, T + 1):
    print(f"#{test_case}", end=" ")
    # 숫자 입력
    n = int(input())
    # 연결 리스트 헤드
    head = None
    # 연결 리스트 꼬리
    tail = None
    # c, id 입력
    for _ in range(n):
        c, i = map(int, input().split())
        # 처음 입력되는 값
        if head is None:
            head = tail = Node(i)
        else:
            # c가 1이면 노드를 생성후 헤드로 설정한 뒤에 전체 연결
            if c == 1:
                node = Node(i)
                node.next = head
                head = node
            # c가 2면 노드를 끝에 생성 후 추가
            else:
                tail.next = Node(i)
                tail = tail.next

    node = head
    # 노드안에 값이 있다면 출력
    while node:
        print(node.data, end=" ")
        node = node.next
    print()

File: week02/박정수/SWEA_27024_줄서기앞뒤삽입.java
import java.util.*;

public class Solution {
	static int N;
	
	static class Node{
		int val;
		Node next;
		
		Node(){
			
		}
		
		Node(int val, Node next){
			this.val = val;
			this.next = next;
		}
	}
	
	public static void main(String args[]) {
		Scanner sc = new Scanner(System.in);
		int T = sc.nextInt();
		
		for(int test_case = 1; test_case <= T; test_case++) {
			N = sc.nextInt();
			
			Node head = new Node();
			
			for(int i = 0; i < N; i++) {
				int c = sc.nextInt();
				int id = sc.nextInt();
				Node node = new Node(id, null);
				if(c == 1) {
					//앞에 추가 
					Node next = head.next;
					head.next = node;
					node.next = next;
				}else {
					//뒤에 노드는 어캐 추적하지
					Node cur = head;
					while(cur.next != null) {
						cur = cur.next;
					}
					cur.next = node;
				}
			}
			
			StringBuilder sb = new StringBuilder();
			while(head.next != null) {
				head = head.next;
				sb.append(head.val);
				sb.append(" ");
			}
			System.out.println("#"+test_case+" " + sb.toString());
		}
	}
}
